examples/records/records.bal

```diff
@@ -0,0 +1,33 @@
+import ballerina/io;
+
+// Here is how you can defines a new record called 'Person'.
+type Person {
+    string name;
+    int age;
+    Person? parent;
+    string status;
+};
+
+function main(string... args) {
+    // Here we create an instance of 'Person' record with an empty record literal. Implicit initial value of each
+    // field type will be assigned to the corresponding field
+    Person p1 = {};
+    io:println(p1);
+
+    // Here we create an instance of the `person` record with a non-empty record literal.
+    Person p2 = {name: "Jack", age: 20, parent: p1};
```

Comment:
Shall we please make this have a space after "{", this is the general convention we used so far. So it would be
"{ name: "Jack", age: 20, parent: p1 };".